extensions/typescript-language-features/src/languageFeatures/inlineHints.ts

```diff
@@ -0,0 +1,99 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as Proto from '../protocol';
+import { DocumentSelector } from '../utils/documentSelector';
+import { ClientCapability, ITypeScriptServiceClient, ServerResponse, ExecConfig } from '../typescriptService';
+import { conditionalRegistration, requireSomeCapability } from '../utils/dependentRegistration';
+import { Range } from '../utils/typeConverters';
+
+namespace ExperimentalProto {
+	export const enum CommandTypes {
+		ProvideInlineHints = 'ProvideInlineHints'
+	}
+
+	export interface ProvideInlineHintsArgs extends Proto.FileRequestArgs {
+		/**
+		 * Start position of the span.
+		 */
+		start: number;
+		/**
+		 * Length of the span.
+		 */
+		length: number;
+	}
+
+	export interface ProvideInlineHintsRequest extends Proto.Request {
+		command: CommandTypes.ProvideInlineHints;
+		arguments: ProvideInlineHintsArgs;
+	}
+
+	interface HintItem {
+		text: string;
+		range: Proto.TextSpan;
+		hoverMessage?: string;
+		whitespaceBefore?: boolean;
+		whitespaceAfter?: boolean;
+	}
+
+	export interface ProvideInlineHintsResponse extends Proto.Response {
+		body?: HintItem[];
+	}
+
+	export interface IExtendedTypeScriptServiceClient {
+		execute<K extends keyof ExtendedTsServerRequests>(
+			command: K,
+			args: ExtendedTsServerRequests[K][0],
+			token: vscode.CancellationToken,
+			config?: ExecConfig
+		): Promise<ServerResponse.Response<ExtendedTsServerRequests[K][1]>>;
+	}
+
+	export interface ExtendedTsServerRequests {
+		'provideInlineHints': [ProvideInlineHintsArgs, ProvideInlineHintsResponse];
+	}
+}
+
+class TypeScriptInlineHintsProvider implements vscode.InlineHintsProvider {
+	constructor(
+		private readonly client: ITypeScriptServiceClient
+	) { }
+
+	async provideInlineHints(model: vscode.TextDocument, range: vscode.Range, token: vscode.CancellationToken): Promise<vscode.InlineHint[]> {
+		const filepath = this.client.toOpenedFilePath(model);
+		if (!filepath) {
+			return [];
+		}
+
+		const start = model.offsetAt(range.start);
```

Comment:
I think you need a call to:

```ts
		await this.formattingOptionsManager.ensureConfigurationOptions(document, options, token);
```

This ensures that the current user settings are synchronized to the TS Server